src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrTests.cs

```diff
@@ -0,0 +1,376 @@
+            Path.Combine(vmrPath, "src", "test-repo", "externals", "external-repo", "additional-file.txt"),
+            Path.Combine(vmrPath, "src", "test-repo", "externals", "external-repo", "external-repo-file.txt"),
+            Path.Combine(vmrPath, "src", "test-repo", "test-repo-file.txt")
+        };
+
+        CheckDirectoryContents(new DirectoryInfo(vmrPath), expectedFiles);
+        await CheckAllIsCommited(vmrPath);
+
+        // Remove submodule
+
+        await RemoveSubmodule(testRepoPath, submodulePath);
+        await CommitAll(testRepoPath, "Remove the submodule");
+        commit = await GetRepoLastCommit(testRepoPath);
+
+        res = await processManager.Execute(darcExecutable, new string[] { "vmr", "update", "--vmr", vmrPath, "--tmp", tmpPath, $"test-repo:{commit}" });
+        res.ExitCode.Should().Be(0);
+
+        expectedFiles = new List<string>
+        {
+            Path.Combine(vmrPath, "git-info", "AllRepoVersions.props"),
+            Path.Combine(vmrPath, "git-info", "test-repo.props"),
+            Path.Combine(vmrPath, "src", "source-manifest.json"),
+            Path.Combine(vmrPath, "src", "source-mappings.json"),
+            Path.Combine(vmrPath, "src", "test-repo", ".gitmodules"),
+            Path.Combine(vmrPath, "src", "test-repo", "test-repo-file.txt")
+        };
+
+        CheckDirectoryContents(new DirectoryInfo(vmrPath), expectedFiles);
+        await CheckAllIsCommited(vmrPath);
+    }
+
+    private void CheckDirectoryContents(DirectoryInfo directory, IList<string> expectedFiles)
+    {
+        var filesInDir = GetAllFilesInDirectory(directory).OrderBy(x => x).ToList();
+        filesInDir.Count.Should().Be(expectedFiles.Count);
+        for(int i = 0; i < expectedFiles.Count; i++)
+        {
+            filesInDir[i].Should().Be(expectedFiles[i]);
+        }
+    }
+
+    private async Task CheckAllIsCommited(string repo)
+    {
+        var gitStatus = await processManager.ExecuteGit(repo, new string[] { "status"}, CancellationToken.None);
+        var statusLog = gitStatus.StandardOutput;
+        statusLog.Should().NotContain("Changes to be committed");
+        statusLog.Should().NotContain("Changes not staged for commit");
+    }
+
+    private ICollection<string> GetAllFilesInDirectory(DirectoryInfo directory)
+    {
+        var files = new List<string>();
+
+        if(directory.Name == ".git")
+        {
+            return files;
+        }
+
+        files.AddRange(directory.GetFiles().Select(f => f.FullName));
+
+        foreach (var subDir in directory.GetDirectories())
+        {
+            files.AddRange(GetAllFilesInDirectory(subDir));
+        }
+
+        return files;
+    }
+
+    private async Task CommitAll(string repo, string commitMsg)
+    {
+        await processManager.ExecuteGit(repo, new string[] { "add", "-A" }, CancellationToken.None);
+        await processManager.ExecuteGit(repo, new string[] { "commit", "-m", commitMsg }, CancellationToken.None);
+    }
```

Comment:
You can do this, if you change `processManager` to be `IProcessManager` I think

```suggestion
    private async Task CommitAll(string repo, string commitMessage)
    {
        await processManager.ExecuteGit(repo, "add", "-A");
        await processManager.ExecuteGit(repo, "commit", "-m", commitMessage);
    }
```